library/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/recorder/DebouncerTest.kt

```diff
@@ -0,0 +1,168 @@
+        val fakeDelayedRunnables = forge.aList(size = forge.anInt(min = 1, max = 10)) { TestRunnable() }
+        val delayInterval = ceil(
+            TEST_MAX_DELAY_THRESHOLD_IN_NS.toDouble() /
+                fakeDelayedRunnables.size.toDouble()
+        ).toLong()
+        val fakeExecutedRunnable = TestRunnable()
+        fakeDelayedRunnables.forEach {
+            testedDebouncer.debounce(it)
+            Thread.sleep(TimeUnit.NANOSECONDS.toMillis(delayInterval))
+        }
+
+        // When
+        testedDebouncer.debounce(fakeExecutedRunnable)
+
+        // Then
+        verify(mockHandler, times(fakeDelayedRunnables.size))
+            .postDelayed(any(), eq(Debouncer.DEBOUNCE_TIME_IN_NS))
+        verify(mockHandler, times(fakeDelayedRunnables.size + 1))
+            .removeCallbacksAndMessages(null)
+        assertThat(fakeDelayedRunnables.filter { it.wasExecuted }.size).isEqualTo(0)
+        assertThat(fakeExecutedRunnable.wasExecuted).isTrue
+    }
+
+    @Test
+    fun `M switch to the handler W debounce { delay threshold reached, more runnables after  }`(
+        forge: Forge
+    ) {
+        // Given
+        val fakeDelayedRunnablesPack1 = forge.aList(size = forge.anInt(min = 1, max = 10)) {
+            TestRunnable()
+        }
+        val fakeDelayedRunnablesPack2 = forge.aList(size = forge.anInt(min = 1, max = 10)) {
+            TestRunnable()
+        }
+        val delayInterval = ceil(
+            TEST_MAX_DELAY_THRESHOLD_IN_NS.toDouble() /
+                fakeDelayedRunnablesPack1.size.toDouble()
+        ).toLong()
+        val fakeExecutedRunnable = TestRunnable()
+        fakeDelayedRunnablesPack1.forEach {
+            testedDebouncer.debounce(it)
+            Thread.sleep(TimeUnit.NANOSECONDS.toMillis(delayInterval))
+        }
+        testedDebouncer.debounce(fakeExecutedRunnable)
+
+        // When
+        fakeDelayedRunnablesPack2.forEach {
+            testedDebouncer.debounce(it)
+        }
+
+        // Then
+        val numOfDelayedInvocations = fakeDelayedRunnablesPack1.size +
+            fakeDelayedRunnablesPack2.size
+        verify(mockHandler, times(numOfDelayedInvocations)).postDelayed(
+            any(),
+            eq(Debouncer.DEBOUNCE_TIME_IN_NS)
+        )
+        val numOfCancelInvocations = fakeDelayedRunnablesPack1.size +
+            fakeDelayedRunnablesPack2.size + 1
+        verify(mockHandler, times(numOfCancelInvocations)).removeCallbacksAndMessages(null)
+        assertThat(fakeDelayedRunnablesPack1.filter { it.wasExecuted }.size).isEqualTo(0)
+        assertThat(fakeDelayedRunnablesPack2.filter { it.wasExecuted }.size).isEqualTo(0)
+        assertThat(fakeExecutedRunnable.wasExecuted).isTrue
+    }
+
+    private class TestRunnable : Runnable {
+        var wasExecuted: Boolean = false
+
+        override fun run() {
+            wasExecuted = true
+        }
+    }
+
+    companion object {
+        private val TEST_MAX_DELAY_THRESHOLD_IN_NS = TimeUnit.SECONDS.toNanos(2)
```

Comment:
minor: this probably can be even lower to make tests faster

Reply:
yup...I hesitated as not to have longer tests, we already have quite a lot but I guess is better than to have flaky ones.